### PR TITLE
chore(tests): ensure tests coverage threshold and hooks run for staged files only

### PR DESCRIPTION
**Pull request by peterkc**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
     "*.js": [
       "prettier --write",
       "eslint",
+      "jest --silent --coverage",
```
Comment (camacho) [inline package.json]: Won't this run for each JS file that's changed? Should we instead configure Husky to run a single test command and *then* lint-staged for the precommit hook.

File: package.json [hunk 2]
```diff
@@ -53,6 +54,14 @@
...
   },
   "jest": {
+    "coverageThreshold": {
+      "global": {
+        "branches": 100,
+        "functions": 100,
+        "lines": 100,
+        "statements": -10
+      }
+    },
```
Comment (camacho) [inline package.json]: Do this match the CodeCov configuration?

File: package.json (hunk 1 above)
Comment (peterkc) [reply package.json]: @camacho Good catch. Have fixed this with the latest push which will only run hooks for staged files only.

File: package.json (hunk 2 above)
Comment (peterkc) [reply package.json]: @camacho I think so. Currently `codecov` becomes an issue when the coverage drops, at the moment we have 100%.

**Comment by peterkc** [comment]
This depends on PR #55 to fix `update-lockfile` status check.

File: package.json [hunk 3]
```diff
@@ -38,13 +38,16 @@
...
+      "./scripts/lint-staged-jest.sh"
+    ],
     "**/package.json": [
       "yarn format:package",
       "git add"
     ],
     "*.js": [
       "prettier --write",
-      "eslint",
+      "./scripts/lint-staged-eslint.sh",
```
Comment (camacho) [inline package.json]: What is the change here? Reading the script code, it seems it is linting all files for each file that changed. I have the same concern about the change above for the tests. My understanding for how `lint-staged` works is that for each changed file that matches the key in the configuration, the array of scripts is run. In this case, the script here would run 1 time per `.js` file. Am I misunderstanding?

Comment (peterkc) [reply package.json]: @camacho My bad, it was me miss understanding lint-staged. Have retested and revert the changes. Good call.

**Comment by peterkc** [comment]
> I'm confused by the changes here and why we didn't update the git hook to run `lint-staged` **and** `jest --coverage` pre-commit

@camacho You are right. `.huskyrc.json` already has the following pre-commit that I added, which takes care of this. 

```
"pre-push": "yarn test --coverage"
```

**Comment by peterkc** [comment]
Have rebased master with the fix for `circleci`, not able to merge without an approval.

**Comment by camacho** [comment]
Blarg. I haven't found a way to give you permissions for this stuff.

**Comment by peterkc** [comment]
> Blarg. I haven't found a way to give you permissions for this stuff.

There might be something we can do with actions after all checks and an approval. I'll add it to the research task in getting semantic-release working.

**Pull request merged**